specification/voiceservices/resource-manager/Microsoft.VoiceServices/cadl/communicationsgateway.cadl

```diff
@@ -90,107 +107,172 @@ model CommunicationsGateway is TrackedResource<CommunicationsGatewayProperties>
 }
 
 @doc("Available platform types.")
-@knownValues(CommunicationsPlatformValues)
-model CommunicationsPlatform is string {}
-enum CommunicationsPlatformValues {
-	OperatorConnect,
-	TeamsPhoneMobile
+enum CommunicationsPlatform {
+  OperatorConnect,
+  TeamsPhoneMobile,
 }
 
 // Currently empty, with presence indicating that this feature should be enabled.
 // This is expected to include more properties in future, hence being an object.
 @doc("Details of API bridge functionality.")
 model ApiBridgeProperties {}
 
+@doc("Available auto-generated domain name scopes.")
+enum AutoGeneratedDomainNameLabelScope {
+    TenantReuse,
+    SubscriptionReuse,
+    ResourceGroupReuse,
+    NoReuse
+}
+
 @doc("Details of the CommunicationsGateway resource.")
 model CommunicationsGatewayProperties is ResourceProperties {
   @doc("The current status of the deployment.")
-  @visibility("read") status?: Status;
+  @visibility("read")
+  status?: Status;
 
   @doc("The regions in which to deploy the resources needed for Teams Calling")
   @OpenAPI.extension("x-ms-identifiers", ["name"])
-  @visibility("read", "write", "create")
+  @visibility("read", "update", "create")
   serviceLocations: ServiceRegionProperties[];
 
   @doc("How to connect back to the operator network, e.g. MAPS")
   @visibility("read", "create")
   connectivity: Connectivity;
 
   @doc("Voice codecs to support")
-  @visibility("read", "write", "create")
+  @visibility("read", "update", "create")
   codecs: TeamsCodecs[];
-  
+
   @doc("How to handle 911 calls")
-  @visibility("read", "write", "create")
+  @visibility("read", "update", "create")
   e911Type: E911Type;
 
   @doc("What platforms to support")
-  @visibility("read", "write", "create")
-  // Can be set once we get a cald release with this decorator
-  // @minItems(1)
+  @visibility("read", "update", "create")
+  @minItems(1)
   platforms: CommunicationsPlatform[];
 
   @doc("Details of API bridge functionality, if required")
-  @visibility("read", "write", "create")
+  @visibility("read", "update", "create")
   apiBridge?: ApiBridgeProperties;
+
+  @added(Versions.v2023_01_31)
+  @doc("The scope at which the auto-generated domain name can be re-used")
+  @visibility("read", "create")
+  autoGeneratedDomainNameLabelScope?: AutoGeneratedDomainNameLabelScope = AutoGeneratedDomainNameLabelScope.TenantReuse;
+
+  @added(Versions.v2023_01_31)
+  @doc("The autogenerated label used as part of the FQDNs for accessing the Communications Gateway")
+  @visibility("read")
+  autoGeneratedDomainNameLabel?: string;
+
+  @added(Versions.v2023_01_31)
+  @doc("This number is used in Teams Phone Mobile scenarios for access to the voicemail IVR from the native dialer.")
+  @visibility("read", "update", "create")
+  teamsVoicemailPilotNumber?: string;
+
+  @added(Versions.v2023_01_31)
+  @doc("Whether an on-premises Mobile Control Point is in use.")
+  @visibility("read", "update", "create")
+  onPremMcpEnabled?: boolean = false;
```

Comment:
>onPremMcpEnabled [](http://example.com/codeflow?start=2&length=16)

Hmm... I don't know how to capitalize onPrem. Is it one word? But maybe it would be obviously correct if we called it onPremisesMcpEnabled?

(How do we know its actually onPrem and not just virtual somehow though?)